BedrockServer.cpp

```diff
@@ -364,11 +364,49 @@ void BedrockServer::sync(const SData& args,
             server._suppressMultiWrite.store(true);
         }
 
-        // If the node's not in a ready state at this point, we'll probably need to read from the network, so start the
-        // main loop over. This can let us wait for logins from peers (for example).
+        // If the node's not in a ready state at this point, we'll probably need to read from the network. Test
+        // conditions from transitional states and then start the main loop over. This can let us wait for logins
+        // from peers (for example).
         if (nodeState != SQLiteNode::LEADING &&
             nodeState != SQLiteNode::FOLLOWING   &&
             nodeState != SQLiteNode::STANDINGDOWN) {
+
+            // If we were LEADING, but we've transitioned, then something's gone wrong (perhaps we got disconnected
+            // from the cluster).
+            if ((preUpdateState == SQLiteNode::LEADING || preUpdateState == SQLiteNode::STANDINGDOWN)
+                     && nodeState != SQLiteNode::LEADING && nodeState != SQLiteNode::STANDINGDOWN) {
```

Comment:
👍 to tyler's comment

Reply:
I agree that the conditional could be different, but I don't follow the suggestion of `nodeState == SQLiteNode::STANDINGDOWN`. nodeState can never by STANDINGDOWN there, because of the conditional at 372. I would propose removing the `nodeState !=` conditionals on line 377, and keeping the `preUpdateState` conditionals on line 376.

Reply:
Oops, I meant `nodeState == SQLiteNode::FOLLOWING`, copy/pasted wrong line.